ui/components/data-sources/AddDataSourceDialog.js

```diff
@@ -0,0 +1,207 @@
+import React, { Component } from "react";
+import {
+    Alert,
+    Modal,
+    Icon,
+    Button,
+    Form,
+    FormControl,
+    FormGroup,
+    Col,
+    InputGroup,
+    DropdownButton,
+    MenuItem
+} from "patternfly-react";
+import some from "lodash.some";
+
+import { graphql } from "react-apollo";
+
+import { InMemoryForms } from "./forms";
+import { DataSourceType } from "../../graphql/types/DataSourceType";
+import CreateDataSource from "../../graphql/CreateDataSource.graphql";
+import GetDataSources from "../../graphql/GetDataSources.graphql";
+
+const INITIAL_STATE = {
+    name: "",
+    type: DataSourceType.InMemory,
+    err: "",
+    inMemoryValues: {
```

Comment:
'InMemoryValues' would suggest that this object is directly related to the In Memory Data Source, but looking at the code below, it looks like this is being used as a generic config object that gets passed along regardless of the data source type.

@josemigallas Let's say we have a Postgres datasource type, would this InMemoryValues object also be used to pass the postgres config (connection details, etc.) back to the API?

Reply:
No, this one is passed only to `InMemoryForms`. When Postgres type is added, there will be a `postgresValues` for the `PostgresForms` component. Everything has to be in the container/parent component tho, as it has the control over the state :(